WordPress/Classes/ViewRelated/NUX/LoginEmailViewController.swift

```diff
@@ -1,35 +1,328 @@
+        super.viewDidAppear(animated)
+
+        assert(SigninHelpers.controllerWasPresentedFromRootViewController(self),
+               "Only present parts of the magic link signin flow from the application's root vc.")
+
+        registerForKeyboardEvents(keyboardWillShowAction: #selector(handleKeyboardWillShow(_:)),
+                                  keyboardWillHideAction: #selector(handleKeyboardWillHide(_:)))
+    }
+
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        unregisterForKeyboardEvents()
     }
 
-    override func signinWithUsernamePassword(_ immediateSignin: Bool = false) {
+
+    // MARK: - Setup and Configuration
+
+
+    /// Let the storyboard's style stay
+    /// TODO: Nuke this and the super implementation once the old signin controllers
+    /// go away. 2017.06.13 - Aerych
+    override func setupStyles() {}
+
+
+    /// Hides the self-hosted login option.
+    ///
+    func configureForWPComOnlyIfNeeded() {
+        selfHostedSigninButton.isHidden = restrictToWPCom
+    }
+
+
+    /// Assigns localized strings to various UIControl defined in the storyboard.
+    ///
+    func localizeControls() {
+        instructionLabel.text = NSLocalizedString("Log in to WordPress.com using an email address to manage all your WordPress sites.", comment: "Instruction text on the login's email addresss screen.")
+
+        emailTextField.placeholder = NSLocalizedString("Email address", comment: "Placeholder for a textfield. The user may enter their email address.")
+        emailTextField.accessibilityIdentifier = "Email address"
+
+        let submitButtonTitle = NSLocalizedString("Next", comment: "Title of a button. The text should be capitalized.").localizedCapitalized
+        submitButton.setTitle(submitButtonTitle, for: UIControlState())
+        submitButton.setTitle(submitButtonTitle, for: .highlighted)
+        submitButton.accessibilityIdentifier = "Next Button"
+
+        let selfHostedTitle = NSLocalizedString("Log into your site by entering your site address instead.", comment: "A button title.")
+        selfHostedSigninButton.setTitle(selfHostedTitle, for: UIControlState())
+        selfHostedSigninButton.setTitle(selfHostedTitle, for: .highlighted)
+        selfHostedSigninButton.titleLabel?.numberOfLines = 0
+    }
+
+
+    /// Sets up a 1Password button if 1Password is available.
+    ///
+    func setupOnePasswordButtonIfNeeded() {
+        WPStyleGuide.configureOnePasswordButtonForTextfield(emailTextField,
+                                                            target: self,
+                                                            selector: #selector(handleOnePasswordButtonTapped(_:)))
+    }
+
+
+    /// Configures the email text field, updating its text based on what's stored
+    /// in `loginFields`.
+    ///
+    func configureEmailField() {
+        emailTextField.textInsets = UIEdgeInsetsMake(7, 20, 7, 20)
```

Comment:
Looks like we'll need to do the same for every text field until we nuke the old UI.  I'll make a style for it.